xorca/lib.py

```diff
@@ -62,10 +62,10 @@ def trim_and_squeeze(ds,
         ds = ds.isel(x=slice(*x_slice))
 
     to_squeeze = [dim for dim in ds.dims
-                  if ((ds[dim].size == 1) and ((dim is not "t")
+                  if (((ds[dim].size == 1) and ((dim is not "t")
```

Comment:
These parens are very hard to parse.

What about:

```python
    to_squeeze = [dim for dim in ds.dims
                  if ((ds[dim].size == 1)
                      and ((dim is not "t") or
                           np.invert(np.issubdtype(ds[dim].dtype,
                                                   np.datetime64)))
                      and (dim not in z_dims))]
```

Reply:
Or, more verbose but self-explaining (check flake for the white space between funcs):

```python
    def _is_singleton(ds, dim):
        return (ds[dim] == 1)

    def _is_time_dim(ds, dim):
        return (dim in orca_names.t_dims or
                np.invert(np.issubdtype(ds[dim].dtype,
                                        np.datetime64)))

    def _is_z_dim(ds, dim):
        return (dim in orca_names.z_dims)

    to_squeeze = [dim for dim in ds.dims
                  if (not _is_singleton(ds, dim) and
                      not _is_time_dim(ds, dim) and
                      not _is_z_dim(ds, dim))]